src/irc/core/irc-servers-setup.c

```diff
@@ -69,7 +69,14 @@ static void sig_server_setup_fill_chatnet(IRC_SERVER_CONNECT_REC *conn,
 		return;
 	g_return_if_fail(IS_IRCNET(ircnet));
 
-	if (ircnet->nick != NULL) g_free_and_null(conn->alternate_nick);
+	if (ircnet->nick != NULL) {
+		g_free_and_null(conn->nick);
+		conn->nick = g_strdup(ircnet->nick);
+	}
+	if (ircnet->alternate_nick != NULL) {
```

Comment:
I think this can be safely moved to `server_setup_fill_chatnet` if it's not irc-specific.
But if it limited to irc-only the whole `alternate_nick` field should go in `_IRC_CHATNET_REC`.